src/PooledResultSet.php

```diff
@@ -18,4 +19,9 @@ public function __construct(ResultSet $result, callable $release)
         parent::__construct($result, $release);
         $this->result = $result;
     }
+
+    public function nextResultSet(): Promise
```

Comment:
Should we rename this to `advanceResultSet()`?

Reply:
Why is this method even needed? We don't have something like it in amp\postgres. Maybe should should hide this somehow as implementation detail that you don't care about and is not part of public api.